lib/src/timeline.dart

```diff
@@ -590,9 +591,15 @@ class Timeline {
 
     // Search on the server
     var prevBatch = room.prev_batch;
+    var ignoreEventIds = <String>[];
     if (sinceEventId != null) {
-      prevBatch =
-          (await room.client.getEventContext(room.id, sinceEventId)).end;
+      final eventContext =
+          await room.client.getEventContext(room.id, sinceEventId);
+      final eventIdsAfter =
+          eventContext.eventsAfter?.map((event) => event.eventId).toList() ??
+              [];
+      ignoreEventIds = eventIdsAfter..add(sinceEventId);
```

Comment:
I think this logic shouldn't be necessary, if you added all the yielded event ids to the `ignoreEventIds` already.